sklearn/preprocessing/imputation.py

```diff
@@ -413,6 +431,21 @@ def transform(self, X):
         valid_idx = np.where(valid_mask)[0]
         missing = np.arange(X.shape[not self.axis])[invalid_mask]
 
+        if sparse.issparse(X) and self.missing_values != 0:
```

Comment:
I tried avoiding this, but the mask needed to be computed before the features with all missing values are discarded since the indicator matrix needs to select the columns based on ``imputed_features_`` which is with respect to the initial array before discarding features. Is there any way I can prevent this ?